src/main/java/pw/itr0/kaba/encrypt/PasswordBasedEncrypter.java

```diff
@@ -12,27 +17,61 @@
+
+    /**
+     * Encrypt {@code secret} based on password.
+     *
      * @param secret   secret bytes
+     * @param password encrypting password
      * @return encrypted secret
-     * @throws GeneralSecurityException on encryption failure
+     * @throws InvalidKeyException       if the {@code password} does not satisfy password specification of using encryption algorithm,
+     *                                   or encryption algorithm using specified password length is not usable on current JDK
+     * @throws BadPaddingException       if the {@code secret} is not properly padded
+     * @throws IllegalBlockSizeException if the {@code secret} has wrong length
      */
-    byte[] encrypt(char[] password, byte[] secret) throws GeneralSecurityException;
+    default byte[] encrypt(byte[] secret, char[] password)
+            throws InvalidKeyException, BadPaddingException, IllegalBlockSizeException {
+        return getEncryptionCipher(password).doFinal(secret);
+    }
 
     /**
      * Encrypt {@code secret} based on password.
      * <p>
      * Encrypted secret is encoded as Base64.
      *
-     * @param password encrypting password
      * @param secret   secret bytes
+     * @param password encrypting password
      * @return encrypted secret (Base64 encoded string)
-     * @throws GeneralSecurityException on encryption failure
+     * @throws InvalidKeyException       if the {@code password} does not satisfy password specification of using encryption algorithm,
+     *                                   or encryption algorithm using specified password length is not usable on current JDK
+     * @throws BadPaddingException       if the {@code secret} is not properly padded
+     * @throws IllegalBlockSizeException if the {@code secret} has wrong length
+     */
+    default String encrypt64(byte[] secret, char[] password)
+            throws InvalidKeyException, BadPaddingException, IllegalBlockSizeException {
+        return Base64.getEncoder().encodeToString(this.encrypt(secret, password));
+    }
+
+    /**
+     * Wrap {@link OutputStream} {@code out} to {@link CipherOutputStream} encrypting with {@code password}.
+     *
+     * @param out      {@link OutputStream} to wrap in {@link CipherOutputStream}
+     * @param password encrypting password
+     * @return encrypting {@link OutputStream}
+     * @throws InvalidKeyException if the {@code password} does not satisfy password specification of using encryption algorithm,
+     *                             or encryption algorithm using specified password length is not usable on current JDK
      */
-    default String encrypt64(char[] password, byte[] secret) throws GeneralSecurityException {
-        return Base64.getEncoder().encodeToString(this.encrypt(password, secret));
+    default OutputStream wrap(OutputStream out, char[] password) throws InvalidKeyException {
+        return new CipherOutputStream(out, getEncryptionCipher(password));
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method call to 'getEncryptionCipher' needs "this.".](https://www.codacy.com/app/ryotan/kaba-vault/file/2686265517/issues/source?bid=3298589&fileBranchId=3332567#l75)